common/services/single-request.js

```diff
@@ -0,0 +1,53 @@
+const { pickBy } = require('lodash')
+
+const moveService = require('./move')
+
+const singleRequestService = {
+  getAll({
+    status,
+    moveDate = [],
+    createdAtDate = [],
+    fromLocationId,
+    toLocationId,
+    isAggregation = false,
+  } = {}) {
+    const [moveDateFrom, moveDateTo] = moveDate
+    const [createdAtFrom, createdAtTo] = createdAtDate
+
+    let statusFilter
+    switch (status) {
+      case 'pending':
+        statusFilter = {
+          'filter[status]': 'proposed',
+        }
+        break
+      case 'accepted':
+        statusFilter = {
```

Comment:
Is there a story here as to why these 3 terms are agglomerated?

Reply:
There is. We are making re-use of the statuses that we know exist in our API. This combines all statuses that signify a request has been "approved" so we want to ensure they will continue to be displayed in the "approved" filter after a move progressing through more states.

Reply:
(but I can see it makes the urls much cleaner ;)

Reply:
In the future we will want to capture changes to state and may be able to derive the "approved" state another way, which I why I wanted to create this service to abstract that.

Currently if a new status was added that suggests a move has been approved we would have had to add it across various places. And also, deriving these particular moves required more filters than just status so also felt it was appropriate to abstract that to this service.